lib/experimental/fonts-library/class-wp-font-family.php

```diff
@@ -0,0 +1,421 @@
+	}
+
+	/**
+	 * Returns the font family data.
+	 *
+	 * @return array An array in fontFamily theme.json format.
+	 */
+	public function get_data() {
+		return $this->data;
+	}
+
+	/**
+	 * Returns the font family data.
+	 *
+	 * @return string fontFamily in theme.json format as stringified JSON.
+	 */
+	public function get_data_as_json() {
+		return wp_json_encode( $this->get_data() );
+	}
+
+	/**
+	 * Returns if the font family has font faces defined.
+	 *
+	 * @return bool true if the font family has font faces defined, false otherwise.
+	 */
+	public function has_font_faces() {
+		return (
+			isset( $this->data['fontFace'] )
+			&& is_array( $this->data['fontFace'] )
+			&& ! empty( $this->data['fontFace'] )
+		);
+	}
+
+	/**
+	 * Removes font family assets.
+	 *
+	 * @return bool True if assets were removed, false otherwise.
+	 */
+	private function remove_font_family_assets() {
+		if ( $this->has_font_faces() ) {
+			foreach ( $this->data['fontFace'] as $font_face ) {
+				$were_assets_removed = $this->delete_font_face_assets( $font_face );
+				if ( false === $were_assets_removed ) {
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Removes a font family from the database and deletes its assets.
+	 *
+	 * @return bool|WP_Error True if the font family was uninstalled, WP_Error otherwise.
+	 */
+	public function uninstall() {
+		$post = $this->get_data_from_post();
+		if ( null === $post ) {
+			return new WP_Error( 'font_family_not_found', __( 'The font family could not be found.', 'gutenberg' ) );
+		}
+		$were_assets_removed = $this->remove_font_family_assets();
+		if ( true === $were_assets_removed ) {
+			$was_post_deleted = wp_delete_post( $post->ID, true );
+			if ( null === $was_post_deleted ) {
+				return new WP_Error( 'font_family_not_deleted', __( 'The font family could not be deleted.', 'gutenberg' ) );
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Deletes a specified font asset file from the fonts directory.
+	 *
+	 * @param string $src The path of the font asset file to delete.
+	 * @return bool Whether the file was deleted.
+	 */
+	private static function delete_asset( $src ) {
+		$filename  = basename( $src );
+		$file_path = path_join( WP_FONTS_DIR, $filename );
+
+		if ( ! file_exists( $file_path ) ) {
+			return false;
+		}
+		wp_delete_file( $file_path );
+
+		// If the file still exists after trying to delete it, return false.
+		if ( file_exists( $file_path ) ) {
+			return false;
+		}
+
+		// return true if the file was deleted.
+		return true;
+	}
+
+	/**
+	 * Deletes all font face asset files associated with a given font face.
+	 *
+	 * @param array $font_face The font face array containing the 'src' attribute with the file path(s) to be deleted.
+	 */
+	private static function delete_font_face_assets( $font_face ) {
+		$srcs = ! empty( $font_face['src'] ) && is_array( $font_face['src'] )
+			? $font_face['src']
+			: array( $font_face['src'] );
+		foreach ( $srcs as $src ) {
```

Comment:
Nitpick: could `$srcs` be replaced with `$sources` or maybe `$font_face_sources` (or a similar name)? I find `$srcs` [a bit hard to read](https://github.com/piotrplenik/clean-code-php#use-meaningful-and-pronounceable-variable-names).
```suggestion
		$sources = ! empty( $font_face['src'] ) && is_array( $font_face['src'] )
			? $font_face['src']
			: array( $font_face['src'] );
		foreach ( $sources as $src ) {
```